packages/palette/src/elements/Box/Box.tsx

```diff
@@ -55,3 +53,28 @@ export const Box = primitives.View<BoxProps>`
 `
 
 Box.displayName = "Box"
+
+/** RegExp for filtering BoxProps */
+export const BOX_PROP_REGEX = new RegExp(`^(${boxMixin.propNames.join("|")})$`)
+
+/** Splits out props into valid and invalid BoxProps */
+export const splitBoxProps = <
+  T extends BoxProps,
+  U extends Omit<T, keyof BoxProps>
+>(
+  props: T
+): [BoxProps, U] => {
+  const boxProps = {} as T
+  const notBoxProps = {} as U
+
+  for (const key of Object.keys(props)) {
+    if (BOX_PROP_REGEX.test(key)) {
+      boxProps[key] = props[key]
+      continue
+    }
+
+    notBoxProps[key] = props[key]
+  }
+
+  return [boxProps, notBoxProps]
+}
```

Comment:
So yeah, the usecase here is that frequently one might want to implement a component in such a way as to split layout concerns with an outer box. This gives you a typesafe way to split off the box related props without having to manually pull them off.